core/templates/dev/head/editor/EditorServices.js

```diff
@@ -2036,3 +2083,58 @@ oppia.factory('explorationWarningsService', [
     };
   }
 ]);
+
+// Service for displaying different types of modals depending on the type of
+// response received as a result of the autosaving request.
+oppia.factory('autosaveInfoModalsService', [
+  '$modal', '$timeout',
+  function(
+    $modal, $timeout) {
+    return {
+      showNonStrictFailModal: function() {
+        $modal.open({
+          templateUrl: 'modals/saveValidationFail',
+          backdrop: true,
+          controller: [
+            '$scope', '$modalInstance',
+            function(
+              $scope, $modalInstance
+            ) {
+              $scope.close = function() {
+                $modalInstance.dismiss('cancel');
+              };
+            }
+          ]
+        }).result.then(function() {
+          // User refreshes the page - Flow for exploration loaded is followed.
+        });
+      },
+      showVersionMismatchModal: function() {
+        $modal.open({
+          templateUrl: 'modals/saveVersionMismatch',
+          backdrop: true,
+          controller: [
+            '$scope', '$modalInstance',
+            function(
+              $scope, $modalInstance
+            ) {
+              $scope.cancel = function() {
+                $modalInstance.dismiss('cancel');
+              };
+              $scope.discardChanges = function() {
+                // Send request to backend to set draft_changes to None.
+                // If the request succeeds ->
```

Comment:
There seem to be various components of this that have not been implemented. Anything committed should be in a working state.